src/Enhavo/Bundle/RoutingBundle/AutoGenerator/Generator/PrefixGenerator.php

```diff
@@ -45,4 +75,38 @@ public function getType()
     {
         return 'prefix';
     }
-}
\ No newline at end of file
+
+    private function getSlug($string)
+    {
+        return Slugifier::slugify($string);
+    }
+
+    /**
+     * $isFirstTry prevents unintended increase. E.g. if user enters "12-12-12" and entry exists, it should become "12-12-12-1", not "12-12-13".
+     */
+    private function getUniqueSlug(string $string, $isFirstTry = true)
+    {
+        $string = $this->getSlug($string);
+
+        $results = $this->routeRepository->findBy([
+            'staticPrefix' => $this->getUrlFormat($string)
+        ]);
+
+        if(count($results)){
+            return $this->getUniqueSlug($this->increase($string, $isFirstTry), false);
+        } else {
+            return $string;
+        }
+    }
+
+    private function increase($string, $isFirstTry)
```

Comment:
`increase` is only called once with `$isFirstTry` `true`. So additional parameter is not necessary. I would remove `$isFirstTry`, If you want to have control over the strategy how a unique slug is extended, than better add a config parameter `unique_strategy` or `unique_method` and offer some strategies like `increase` or `append`